WordPress/src/main/java/org/wordpress/android/ui/media/MediaSettingsActivity.java

```diff
@@ -745,32 +948,76 @@ public void onReceive(Context context, Intent intent) {
     private void saveChanges() {
         if (isFinishing()) return;
 
-        MediaModel media = mMediaStore.getMediaWithLocalId(mMedia.getId());
-        if (media == null) {
-            AppLog.w(AppLog.T.MEDIA, "MediaSettingsActivity > Cannot save null media");
-            ToastUtils.showToast(this, R.string.media_edit_failure);
-            return;
-        }
-
         String thisTitle = EditTextUtils.getText(mTitleView);
         String thisCaption = EditTextUtils.getText(mCaptionView);
         String thisAltText = EditTextUtils.getText(mAltTextView);
         String thisDescription = EditTextUtils.getText(mDescriptionView);
 
-        boolean hasChanged = !StringUtils.equals(media.getTitle(), thisTitle)
-                || !StringUtils.equals(media.getCaption(), thisCaption)
-                || !StringUtils.equals(media.getAlt(), thisAltText)
-                || !StringUtils.equals(media.getDescription(), thisDescription);
-        if (hasChanged) {
-            AppLog.d(AppLog.T.MEDIA, "MediaSettingsActivity > Saving changes");
-            media.setTitle(thisTitle);
-            media.setCaption(thisCaption);
-            media.setAlt(thisAltText);
-            media.setDescription(thisDescription);
-            mDispatcher.dispatch(MediaActionBuilder.newPushMediaAction(new MediaStore.MediaPayload(mSite, media)));
+        if (!isMediaFromEditor()) {
```

Comment:
Just a heads-up that at some point we'll want to save changes to the media store even when shown from the editor. So, for example, if the user inserts an image from the WP media library and then edits the alt text, we'll want to save the alt text back to the store. That's outside the scope of this PR, though.